python/ray/serve/backend_worker.py

```diff
@@ -90,6 +90,9 @@ async def __init__(self, backend_tag, replica_tag, init_args,
                                            backend_config.user_config, version,
                                            is_function, controller_handle)
 
+            # asyncio.Event to be used to signal replica enterred shutdown phase.
+            self.shutdown_event = asyncio.Event()
+
         @ray.method(num_returns=2)
         async def handle_request(
```

Comment:
what will happen here if shutdown event is set ? Same for reconfigure() underneath. Should we stop taking new calls into the replica ?

Reply:
The shutdown event is only set when the replica enter STOPPING state. It will process requests until graceful shutdown timeout. During this time, their handles should be removed from all the client.